### ISSUE TITLE
jQuery-ujs support

### ISSUE THREAD

**Issue by JamesPoel** [mistic100/Bootstrap-Confirmation#86]
Can this be updated to support jQuery-ujs. The click handler for that is `click.rails` but this only triggers `click` - as such, the actions are being submitted through `GET` instead of `DELETE`.

**Comment by JamesPoel** [comment]
For anyone experiencing a similar issue, my current workaround is using jquery-ujs and the following:
```
$( document ).on( 'click', '[data-apply=confirmation]', function ( e ) {
    var id = $( this ).closest( '.popover.confirmation' ).attr( 'id' );
    $( '[aria-describedby=' + id + ']' ).trigger( 'click.rails' );
    e.preventDefault();
} );
```

**Comment by mistic100** [comment]
Please provide an example of your usage. 

Because the confirmation doesn't do anything on itself, it just blocks the click event and triggers a new click event when the Ok button clicked.

**Comment by JamesPoel** [comment]
I am using [jquery-ujs](https://github.com/rails/jquery-ujs) as my delete button, which is like this:
`<a data-method="delete" href="...">Delete</a>`
When clicked, rails submits the action as a delete request - all OK.
With the confirmation plugin enabled, it submits the request as GET request.
I'm not 100% on how ujs works, but it appears that its on click event is `click.rails` and not simply `click`.
Perhaps it would be possible to check if this event exists, and if so, trigger it, if not, fallback to click event?

**Comment by mistic100** [comment]
This is ultra specific, I think your solution is the best.

**Comment by mistic100** [comment]
Or I can add a "confirmationEvent" option, by default it will be "click" (or the value of "trigger" for BT4) and is customisable.

**Comment by JamesPoel** [comment]
Works for me :)

**Comment by mistic100** [comment]
Available in 2.4.2 and 4.0.1

**Comment by juanmoyano** [comment]
It seems that this change breaks functionality with bootstrap v3.3.7 and jQuery v3.3.1.
If I don't explicitly add confirmationEvent: 'click', it doesn't trigger the click event.
The markup for the element is:
`<button type="button" class="btn btn-xs btn-default" data-toggle="confirmation"></button>`
